build-tracker.config.js

```diff
@@ -14,4 +26,100 @@ module.exports = {
     'dist/dt-report-resources/report.js',
     'dist/dt-report-resources/report-generator.js',
   ],
+  onCompare: compareBuildsAndReportGhStatus,
 };
+
+
+// Thanks to Oliver J Ash! https://github.com/paularmstrong/build-tracker/issues/199
+function compareBuildsAndReportGhStatus(data) {
+  const {comparatorData} = data;
+  const comparator = Comparator.deserialize(comparatorData);
+
+  // Nice table display of the results and diff
+  console.log(comparator.toMarkdown());
+
+  if (comparator.builds.length === 1) {
+    // We need 2 builds to provide a useful comparison, so if there is only one build, we exit early.
+    return Promise.reject(
+      new Error(
+        `Expected comparator to have exactly 2 builds, but it only has 1. ` +
+        `This most likely means the base branch hasn't finished building yet. ` +
+        `Retry this build when the base branch has finished building. ` +
+        `Note you will need to push a new commit.`
+      )
+    );
+  }
+
+  const formatDelta = allArtifactDelta =>
+  `${formatBytes(allArtifactDelta.sizes[SIZE_KEY])} (${(
+    allArtifactDelta.percents[SIZE_KEY] * 100
+  ).toFixed(1)}%)`;
+
+  const getSummaryStats = (parentBuild, build) => {
+    const parentBuildSum = parentBuild.getSum(parentBuild.artifactNames);
+    const buildSum = build.getSum(build.artifactNames);
+    const allArtifactDelta = new ArtifactDelta('All', [], buildSum, parentBuildSum, false);
+    const formattedDelta = formatDelta(allArtifactDelta);
+    const formattedTotalSize = formatBytes(buildSum[SIZE_KEY]);
+    const summaryStats = [
+      `Change: ${formattedDelta}`,
+      `Total: ${formattedTotalSize}`,
+    ].join(' ');
+    return summaryStats;
+  };
+
+  const parentBuild = comparator.builds[0];
+  const build = comparator.builds[1];
+  const targetUrl =
+    `${applicationUrl}/builds/` +
+    `${build.getMetaValue('parentRevision')}/${build.getMetaValue('revision')}`;
+
+  const summaryStats = getSummaryStats(parentBuild, build);
+
+  const isSuccess = comparator.errors.length === 0;
+  const hasWarnings = comparator.unexpectedHashChanges.length || comparator.warnings.length;
+
+  const summary = [hasWarnings ? '⚠️  See report.' : undefined, summaryStats]
+    .filter(s => s !== undefined)
+    .join(' ');
+
+  console.log(`\n${summary}`);
+
+  return postGHStatusForSha(
+    {
+      // Note: we don't use the `build` revision because—in a pull request build—this will
```

Comment:
this is a comment about not using the build revision, but it looks like we are? is `build.meta.revision` not the "`build` revision"?